network/memsocket/src/lib.rs

```diff
@@ -305,11 +315,11 @@ impl MemorySocket {
         let (socket_a, socket_b) = Self::new_pair();
         // Send the socket to the listener
         if let Err(e) = sender.unbounded_send(socket_a) {
-            if e.is_disconnected() {
-                return Err(ErrorKind::AddrNotAvailable.into());
+            if !e.is_disconnected() {
+                error!("MUST_FIX e should not be disconnected");
             }
+            return Err(ErrorKind::AddrNotAvailable.into());
```

Comment:
I'm not sure this message adds value?  If I see this message in the logs, what is `e` without then looking deep into the code?

Reply:
We need to be very careful when making changes like this. This code was constructed under the assumption that only an error of `is_disconnected` is possible. blindly changing this without verifying we can handle other error cases would be a mistake.